src/init/init.ts

```diff
@@ -112,11 +118,7 @@ function getGeneratorDescription(generator: YeomanEnvironment.GeneratorMeta, gen
  * Extract the meaningful name from the full yeoman generator name
  */
 function getDisplayName(generatorName: string) {
-  let nameEnd = generatorName.indexOf(':');
-  if (nameEnd === -1) {
-    nameEnd = generatorName.length;
-  }
-  return generatorName.substring('polymer-init-'.length, nameEnd);
+  return generatorName.replace(/(generator-)?polymer-init-([^:]+)(:.*)?/g, '$2');
```

Comment:
> Doesn't this replace the whole string with one group?

Yep. Example:

```
getDisplayName('generator-polymer-init-foo:aaa')  === 'foo'
getDisplayName('polymer-init-bar:bbb')            === 'bar'
getDisplayName('this-is-my-super-cool-name:ccc')  === 'this-is-my-super-cool-name'
getDisplayName('generator-polymer-init-foo')      === 'foo'
getDisplayName('polymer-init-bar')                === 'bar'
getDisplayName('this-is-my-super-cool-name')      === 'this-is-my-super-cool-name'
```

> If so, I don't think replace() is necessary, and you could just return the group.

Ok, I think you're saying you prefer this: `/(generator-)?polymer-init-([^:]+)(:.*)?/g.exec('generator-polymer-init-foo-el')[2]`. Let me know if I'm wrong here.

The reason I went with `String.replace()` is because it accepts strings that don't match the pattern (i.e., if the regex doesn't pass, it replaces nothing and returns the original string), which allows more flexibility in the display name. `RegExp.exec(str)` returns `null` if no match found, so it would need a little bit of extra code to safely get the capture group.